src/utils/log.rs

```diff
@@ -0,0 +1,27 @@
+use std::{
+    fs::{create_dir_all, File},
+    process::exit,
+};
+
+use chrono::Local;
+use dirs::home_dir;
+
+pub fn initialize_log_file() -> File {
+    let directory = match home_dir() {
+        Some(dir) => dir,
+        None => {
+            println!("Failed to get home directory.");
+            exit(-1);
+        }
+    }
+    .join(".local")
+    .join("share")
+    .join("todui")
+    .join("logs");
```

Comment:
Can these join statements be concatenated into one statement ? Also, it would be nice to have the hard-coded string values as a constant variable.